### ISSUE TITLE
generateNegativeControlOutcomeCohorts does not pass tempEmulationSchema properly

### ISSUE THREAD

**Issue by anthonysena** [OHDSI/CohortGenerator#99]
`generateNegativeControlOutcomeCohorts` has a parameter to set the `tempEmulationSchema` but it is not properly passed to the `insertTable` function:

https://github.com/OHDSI/CohortGenerator/blob/30fe75d61cc8414f6021786502410fe7f66a3ac2/R/NegativeControlCohorts.R#L132-L143

**Comment by anthonysena** [comment]
@ablack3 - I was digging into this some more: since while I am not explicitly passing the `tempEmulationSchema`, DatabaseConnector will default to use `getOption("sqlRenderTempEmulationSchema")` so digging into things a bit more I cam across this condition in `DatabaseConnector::insertTable.default`:

```r
  useCtasHack <- dbms %in% c("pdw", "redshift", "bigquery", "hive") && createTable && nrow(data) > 0 && !useBulkLoad
  if (dbms == "bigquery" && useCtasHack && is.null(tempEmulationSchema)) {
    abort("tempEmulationSchema is required to use insertTable with bigquery when inserting into a new table")
  }
```

The `useCtasHack` variable is set based on a few criteria: the only one I wasn't sure of in this case is the `useBulkLoad` which is specified by `bulkLoad = Sys.getenv("DATABASE_CONNECTOR_BULK_UPLOAD")`. All the other criteria in that boolean expression should be TRUE: you are using BigQuery, we are explicitly setting createTable = TRUE and nrow(data) > 0 since there are > 0 negative controls specified in the SOS challenge. Can you confirm if Sys.getenv("DATABASE_CONNECTOR_BULK_UPLOAD") is set for you?

**Comment by ablack3** [comment]
Thanks Sena! DATABASE_CONNECTOR_BULK_UPLOAD is not currently set.
Should I set this to FALSE in my .Renviron?
`DATABASE_CONNECTOR_BULK_UPLOAD=FALSE`

**Comment by anthonysena** [comment]
Could you give that a try @ablack3? This would simply be a work-around. I may need to refactor the method for negative controls so that we don't have to do the insert into the DB.

**Comment by ablack3** [comment]
Yep I just tried it and it still fails with the same error. I posted the log in our Teams conversation.

**Comment by ablack3** [comment]
I tried a simple test by running this code on bigquery which worked fine. So it seems to me like the sqlRenderTempEmulationSchema option is not being picked up inside the module but I'm not really sure.

```
library(Strategus)
library(DatabaseConnector)
options(sqlRenderTempEmulationSchema =  "main.scratch")

cd <- retrieveConnectionDetails("test", keyringName = "sos")

connection <- connect(cd)

# test insert temp table
insertTable(
   connection = connection, 
   data = cars, 
   tableName = "#test", 
   camelCaseToSnakeCase = TRUE, 
   dropTableIfExists = TRUE, 
   createTable = TRUE, 
   tempTable = TRUE)

n <- renderTranslateQuerySql(con, "select count(*) as n from #test")[[1]]

if (n == nrow(cars)) { 
    message("write access verified")
} else {
    stop("problem with write access")
}

disconnect(con)

```

**Comment by ablack3** [comment]
One thing you might consider adding to the module is an initial test for write access of both temp and persistent tables. If that test fails then no need to continue study execution. I imagine some modules will not need write access.
